### PR TITLE
EI-1205 move incentive payment profiles out of configuration

### PR DESCRIPTION
**Pull request by David-B-Read**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/TestWebApi.cs [hunk 1]
```diff
@@ -41,8 +41,7 @@ public TestWebApi(TestContext context, IHook<object> eventMessageHook, IHook<ICo
                     { "ConfigNames", "SFA.DAS.EmployerIncentives" }
                 };
 
-            _paymentProfiles = new IncentivePaymentProfileListBuilder().Build();
-
+            _paymentProfiles = new IncentivePaymentProfilesService().Get().GetAwaiter().GetResult();
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/TestWebApi.cs:44]: dis doesn't look good

Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/TestWebApi.cs:44]: can't we just inject the result set ready for action?

Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Api.AcceptanceTests/TestWebApi.cs:44]: Looks like _paymentProfiles wasn't being used anyway

File: src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ApprenticeshipIncentive/WhenCalculateEarningsInPhase2.cs [hunk 2]
```diff
@@ -9,11 +9,11 @@
 using SFA.DAS.EmployerIncentives.Domain.Interfaces;
 using SFA.DAS.EmployerIncentives.Domain.ValueObjects;
 using SFA.DAS.EmployerIncentives.Enums;
-using SFA.DAS.EmployerIncentives.UnitTests.Shared.Builders;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.UnitTests.Shared.Builders;
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ApprenticeshipIncentive/WhenCalculateEarningsInPhase2.cs:16]: is this actually needed?

File: src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ValueObjects/WhenCreatingTheIncentive.cs [hunk 3]
```diff
@@ -4,11 +4,11 @@
 using SFA.DAS.EmployerIncentives.Domain.Interfaces;
 using SFA.DAS.EmployerIncentives.Domain.ValueObjects;
 using SFA.DAS.EmployerIncentives.Enums;
-using SFA.DAS.EmployerIncentives.UnitTests.Shared.Builders;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.UnitTests.Shared.Builders;
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ValueObjects/WhenCreatingTheIncentive.cs:11]: is this actually needed?

File: src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ValueObjects/WhenGeneratingEarnings.cs [hunk 4]
```diff
@@ -5,11 +5,11 @@
 using SFA.DAS.EmployerIncentives.Domain.Interfaces;
 using SFA.DAS.EmployerIncentives.Domain.ValueObjects;
 using SFA.DAS.EmployerIncentives.Enums;
-using SFA.DAS.EmployerIncentives.UnitTests.Shared.Builders;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.UnitTests.Shared.Builders;
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ValueObjects/WhenGeneratingEarnings.cs:12]: is this actually needed?

File: src/SFA.DAS.EmployerIncentives.Commands/Services/IncentivePaymentProfilesService.cs [hunk 5]
```diff
@@ -1,39 +1,37 @@
...
     {
-        private readonly ApplicationSettings _applicationSettings;
-
-        public IncentivePaymentProfilesService(IOptions<ApplicationSettings> applicationSettings)
-        {
-            _applicationSettings = applicationSettings.Value;
-        }
-
-        public Task<IEnumerable<Domain.ValueObjects.IncentivePaymentProfile>> Get()
+        public Task<IEnumerable<IncentivePaymentProfile>> Get()
```
Comment (ruslanss) [inline src/SFA.DAS.EmployerIncentives.Commands/Services/IncentivePaymentProfilesService.cs:12]: not much point it being async if we don't utilise it. i know it was that before but we might as well tidy it up now.

**Review by ruslanss — APPROVED** [review]

File: src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ApprenticeshipIncentive/WhenCalculateEarningsInPhase2.cs (hunk 2 above)
Comment (David-B-Read) [reply src/SFA.DAS.EmployerIncentives.Domain.UnitTests/ApprenticeshipIncentive/WhenCalculateEarningsInPhase2.cs:16]: It's the namespace where the unit test utility class IncentivePaymentProfileListBuilder resides. There was a separate utility class under the SFA.DAS.EmployerIncentives.UnitTests.Shared.Builders.Configuration namespace, but that has now been removed.

File: src/SFA.DAS.EmployerIncentives.Commands/Services/IncentivePaymentProfilesService.cs [hunk 6]
```diff
@@ -1,39 +1,35 @@
...
+                    }),
+                new IncentivePaymentProfile(new IncentivePhase(Phase.Phase2),
+                    new List<PaymentProfile>
+                    {
+                        new PaymentProfile(IncentiveType.UnderTwentyFiveIncentive, daysAfterApprenticeshipStart: 89, amountPayable: 1500),
+                        new PaymentProfile(IncentiveType.UnderTwentyFiveIncentive, daysAfterApprenticeshipStart: 364, amountPayable: 1500),
+                        new PaymentProfile(IncentiveType.TwentyFiveOrOverIncentive, daysAfterApprenticeshipStart: 89, amountPayable: 1500),
+                        new PaymentProfile(IncentiveType.TwentyFiveOrOverIncentive, daysAfterApprenticeshipStart: 364, amountPayable: 1500)
+                    }),
+            };
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerIncentives.Commands/Services/IncentivePaymentProfilesService.cs:32]: These need to be in the phase 1 and phase 2 incentive classes, respectively.  That way, everything about the scheme/phase is contained in a single place. IncentivePaymentProfilesService would then be redundant.

**Review by MichaelYoung1981 — APPROVED** [review]

**Pull request merged**